torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -144,69 +226,15 @@ class ScriptModuleSerializer {
     }
   }
 
-  static IValue Tup(std::vector<IValue> ivalues) {
-    return c10::ivalue::Tuple::create(std::move(ivalues));
-  }
-  static IValue Table(std::vector<std::pair<std::string, IValue>> entries) {
-    std::vector<IValue> ivalue_entries;
-    for (const auto& e : entries) {
-      ivalue_entries.push_back(Tup({e.first, e.second}));
-    }
-    return Tup(std::move(ivalue_entries));
-  }
-
   void writeByteCode(const script::Module& module) {
-    auto methods = module.get_methods();
     std::vector<c10::IValue> elements;
-    for (const auto& method : methods) {
-      const auto& func = method.function();
-      auto graph = func.graph()->copy();
-      Inline(*graph);
-      torch::jit::Code code(graph);
-
-      // operator names
-      std::vector<c10::OperatorName> opnames;
-      for (size_t i = 0; i < code.instructions().size(); ++i) {
-        Instruction ins = code.instructions()[i];
-        if (ins.op == OP || ins.op == OPN) {
-          auto node = code.instructions_source()[i];
-          opnames.emplace_back(node->schema().operator_name());
-        }
-      }
 
-      // instructions
-      std::vector<IValue> instructions;
-      for (Instruction ins : code.instructions()) {
-        instructions.emplace_back(Tup({toString(ins.op), ins.X, ins.N}));
-      }
-
-      // operators
-      std::vector<IValue> operators;
-      for (const auto& opname : opnames) {
-        operators.emplace_back(c10::ivalue::Tuple::create({opname.name, opname.overload_name}));
-      }
-
-      // constants
-      auto constants = code.constant_table();
-
-      // types
-      std::vector<IValue> types;
-      for (const TypePtr& t : code.type_table()) {
-        types.push_back(t->python_str());
-      }
-
-      // since the register location is embedded into the bytecode, pass the register size
-      auto register_size = static_cast<int>(code.register_size());
-
-      auto table = Table({{"instructions", Tup(instructions)},
-                          {"operators", Tup(operators)},
-                          {"constants", Tup(constants)},
-                          {"types", Tup(types)},
-                          {"register_size", register_size}});
-
-      elements.push_back(Tup({func.qualname().qualifiedName(), table}));
-    }
-    writeArchive("bytecode", Tup(std::move(elements)));
+    // top-level methods
```

Comment:
This approach will not find all the setstate methods that need to be saved. For instance, if there is a custom class as an attribute of a Module, this will not save it. The correct approach is already implemented for standard serialization: it collects all the classes/functions that have been referenced and makes sure to save them. This code should reuse the result of that analysis to know what __setstate__ methods to produce.

Reply:
Good catch! I've changed it the similar way as in `pickler`, to check `ivalue` of the `module` and its `slots`. `checkHasValidSetGetState()` is reused.